native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -485,8 +537,41 @@ private static BArray newJsonList() {
         return ValueCreator.createArrayValue(JSON_ARRAY_TYPE);
     }
 
-    private static BMap<BString, Object> newJsonMap() {
-        return ValueCreator.createMapValue(TypeCreator.createMapType(JSON_MAP_TYPE));
+    private static BMap<BString, Object> newMap(Type type) {
+        if (type != null) {
+            if (type.getTag() == TypeTags.MAP_TAG) {
+                Type valueType = ((MapType) type).getConstrainedType();
+                switch (valueType.getTag()) {
+                    case TypeTags.STRING_TAG:
+                        return ValueCreator.createMapValue(STRING_MAP_TYPE);
```

Comment:
Can't we just pass the valueType to `ValueCreator.createMapValue(valueType)` instead of handling each type separately